src/h5web/visualizations/shared/utils.ts

```diff
@@ -212,3 +213,14 @@ export function isScaleType(val: unknown): val is ScaleType {
 
   return Object.values<string>(ScaleType).includes(val);
 }
+
+export function lineSegment(
+  startX: number,
+  startY: number,
+  endX: number,
+  endY: number
+): [Vector3, Vector3] | [] {
+  return [startX, startY, endX, endY].every(isFinite)
+    ? [new Vector3(startX, startY, 0), new Vector3(endX, endY, 0)]
```

Comment:
It's a shame that `DataCurve` doesn't benefit from this `every(isFinite)`. Perhaps what's needed is more a helper to make a `Vector3` from two values? This would avoid having to destructure the result of `lineSegment`... but then you'd have to filter for `undefined` afterwards. Hmm... Not sure.

Reply:
The aim of this function is to provide segments for `LineSegment` geometry` while `DataCurve` uses the `Line` geometry`. The difference is that `Line` draws segments between successive points while `LineSegments` draws segments between couples of points.

To use this in `DataCurve`, I would have to switch to `LineSegment`. It is possible but I would have to put explicitly the segments between successive points (i.e. duplicating points).

Reply:
I was more suggesting replacing `lineSegment` with `makePoint` or similar (i.e. to return a single `Vector`) so that you could use it in `DataCurve` without changing the geometry. In `ErrorBarCurve`, you would then call the functiion twice per line. But I'm not convinced this simplies things, so nevermind 😉

Reply:
Yeah and also, it changes the behaviour I wanted: if one of the point of the segment has not finite coordinates, I want to skip[ to skip the whole segment and not only the point in question.